### ISSUE TITLE
Apple Music fails with Got status 500

### ISSUE THREAD

**Issue by databus23** [jishi/node-sonos-http-api#663]
Trying to to play from apple music (e.g. `/RoomName/applemusic/now/song:1143568907`) does not work. While I get `{"status":"success"}`its not starting to play the requested song.

in the log I can see:
```
  Error: Got status 500 when invoking /MediaRenderer/AVTransport/Control
    at Object.invoke (/app/node_modules/sonos-discovery/lib/helpers/soap.js:99:10)
    at Player.nextTrack (/app/node_modules/sonos-discovery/lib/models/Player.js:466:15)
    at player.coordinator.addURIToQueue.then (/app/lib/actions/appleMusic.js:52:62)
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)
```

Running with `NODE_LOG_LEVEL=trace` I can see this:
```
2018-11-23T23:11:49.350Z TRACE http://10.0.4.75:1400/MediaRenderer/AVTransport/Control urn:schemas-upnp-org:service:AVTransport:1#AddURIToQueue <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:AddURIToQueue xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><EnqueuedURI>x-sonos-http:song%3A1143568907.mp4?sid=204&amp;flags=8224&amp;sn=4</EnqueuedURI><EnqueuedURIMetaData>&lt;DIDL-Lite xmlns:dc=&quot;http://purl.org/dc/elements/1.1/&quot; xmlns:upnp=&quot;urn:schemas-upnp-org:metadata-1-0/upnp/&quot;
  xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot; xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&quot;&gt;
  &lt;item id=&quot;00032020song%3A1143568907&quot; parentID=&quot;0004206calbum%3a&quot; restricted=&quot;true&quot;&gt;&lt;dc:title&gt;&quot;&quot;&lt;/dc:title&gt;&lt;upnp:class&gt;object.item.audioItem.musicTrack&lt;/upnp:class&gt;
  &lt;desc id=&quot;cdudn&quot; nameSpace=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot;&gt;SA_RINCON52231_X_#Svc52231-0-Token&lt;/desc&gt;&lt;/item&gt;&lt;/DIDL-Lite&gt;</EnqueuedURIMetaData><DesiredFirstTrackNumberEnqueued>2</DesiredFirstTrackNumberEnqueued><EnqueueAsNext>1</EnqueueAsNext></u:AddURIToQueue></s:Body></s:Envelope>
2018-11-23T23:11:50.120Z TRACE status code: 200
2018-11-23T23:11:50.123Z TRACE
  { firsttracknumberenqueued: '2',
  numtracksadded: '1',
  newqueuelength: '215' }
2018-11-23T23:11:50.125Z TRACE http://10.0.4.75:1400/MediaRenderer/AVTransport/Control urn:schemas-upnp-org:service:AVTransport:1#Next <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:Next xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID></u:Next></s:Body></s:Envelope>
2018-11-23T23:11:50.126Z DEBUG invoking play
2018-11-23T23:11:50.126Z TRACE http://10.0.4.75:1400/MediaRenderer/AVTransport/Control urn:schemas-upnp-org:service:AVTransport:1#Play <s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body><u:Play xmlns:u="urn:schemas-upnp-org:service:AVTransport:1"><InstanceID>0</InstanceID><Speed>1</Speed></u:Play></s:Body></s:Envelope>
2018-11-23T23:11:50.133Z ERROR
  Error: Got status 500 when invoking /MediaRenderer/AVTransport/Control
    at Object.invoke (/app/node_modules/sonos-discovery/lib/helpers/soap.js:99:10)
    at Player.nextTrack (/app/node_modules/sonos-discovery/lib/models/Player.js:466:15)
    at player.coordinator.addURIToQueue.then (/app/lib/actions/appleMusic.js:52:62)
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)
```

Any suggestions on how to get this working? Thanks!

**Comment by polarbeared** [comment]
I have exactly the same issue.

**Comment by jishi** [comment]
Hi, does this only fail some times (especially, from a freshly booted player)? 

I'm wondering if it adds to the queue but doesn't select the queue as the source. Could be verified by manually starting playback once (from the queue, in the app), then pausing and trying to add an apple track again. 

I don't have apple music myself, so it's harder for me to troubleshoot this.

**Comment by databus23** [comment]
@jishi Seems you are on to something. When I played something from Apple music before I try to use the api it works. When I play something else (Radio) and use the API I get the above error and the radio from before starts playing again.

**Comment by jishi** [comment]
Sounds reasonable, radio stations doesn't use the queue. Thanks for
confirming it.

**Comment by jishi** [comment]
Fixed in 1.6.6, but I have no means for testing it. Would be great if someone could verify it!

**Comment by databus23** [comment]
@jishi Works  for me. Thanks a lot!
